### ISSUE TITLE
Support for external variables reference before deployment

### ISSUE THREAD

**Issue by yacut** [localstack/serverless-localstack#65]
### Background
I would like to use external variables like `${ssm:abc}` or `${cf:def}` in my `serverless.yml` file so I can reference values per stage. More info here: https://serverless.com/framework/docs/providers/aws/guide/variables/#reference-variables-using-the-ssm-parameter-store

### Expected behavior
`serverless deploy --stage local` command reconfigures all service endpoints, resolves the external variables and then deploys the functions

### Actual behavior
`serverless deploy --stage local` command tries to resolve the external variables and fails before deployment

``` 
Serverless Warning --------------------------------------

  A valid SSM parameter to satisfy the declaration 'ssm:abc' could not be found.
```

**Comment by whummer** [comment]
Thanks for reporting @matt-jenner , this should be fixed in #67 . Can you please give it a try with the latest version (`0.4.23`). Please report here if the problem persists. Thanks

**Comment by yacut** [comment]
@whummer Thank you for your work. I think the plugin works the right way now, but unfortunately  only for local stage. I get another error when I try to deploy to non-local stage:
```
Serverless: config.options_stage: dev 
Serverless: serverless.service.custom.stage: undefined 
Serverless: serverless.service.provider.stage: ${opt:stage, "dev"} 
Serverless: config.stage: dev 
Serverless: config.options_stage: dev 
Serverless: serverless.service.custom.stage: undefined 
Serverless: serverless.service.provider.stage: dev 
Serverless: config.stage: dev 
Serverless: config.options_stage: dev 
Serverless: serverless.service.custom.stage: undefined 
Serverless: serverless.service.provider.stage: dev 
Serverless: config.stage: dev 
Serverless: WarmUp: no functions to warm up 
Serverless: Packaging service... 
Serverless: Excluding development dependencies... 
Serverless: Excluding development dependencies... 
Serverless: Excluding development dependencies... 
Serverless: Excluding development dependencies... 
Serverless: Excluding development dependencies... 
Serverless: Excluding development dependencies... 
Serverless: Excluding development dependencies... 
Serverless: Excluding development dependencies... 
  Serverless Error --------------------------------------- 
  Could not locate deployment bucket. Error: Cannot read property 'getCredentials' of undefined 
  Get Support -------------------------------------------- 
     Docs:          docs.serverless.com 
     Bugs:          github.com/serverless/serverless/issues 
     Issues:        forum.serverless.com 
  Your Environment Information --------------------------- 
     Operating System:          linux 
     Node Version:              10.16.3 
     Framework Version:         1.60.4 
     Plugin Version:            3.2.6 
     SDK Version:               2.2.1 
     Components Core Version:   1.1.2 
     Components CLI Version:    1.4.0 
```
